testing/scenario_app/ios/Scenarios/ScenariosUITests/PlatformViewUITests.m

```diff
@@ -0,0 +1,118 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/flutter.h>
+#import <XCTest/XCTest.h>
+#include <sys/sysctl.h>
+
+#import "../Scenarios/TextPlatformView.h"
+
+@interface PlatformViewUITests : XCTestCase
+@property(nonatomic, strong) XCUIApplication* application;
+@end
+
+@implementation PlatformViewUITests
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--platform-view" ];
+  [self.application launch];
+}
+
+- (void)testPlatformView {
+  NSBundle* bundle = [NSBundle bundleForClass:[self class]];
+  NSString* goldenName =
+      [NSString stringWithFormat:@"golden_platform_view_%@", [self platformName]];
+  NSString* path = [bundle pathForResource:goldenName ofType:@"png"];
+  UIImage* golden = [[UIImage alloc] initWithContentsOfFile:path];
+
+  XCUIScreenshot* screenshot = [[XCUIScreen mainScreen] screenshot];
+  if (golden) {
+    XCTAttachment* goldenAttachment = [XCTAttachment attachmentWithImage:golden];
+    goldenAttachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [self addAttachment:goldenAttachment];
+  }
+
+  XCTAttachment* attachment = [XCTAttachment attachmentWithScreenshot:screenshot];
+  attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+  [self addAttachment:attachment];
+  XCTAssertTrue([self compareImage:golden toOther:screenshot.image]);
+}
+
+- (NSString*)platformName {
+  NSString* simulatorName =
+      [[NSProcessInfo processInfo].environment objectForKey:@"SIMULATOR_DEVICE_NAME"];
+  if (simulatorName) {
+    return [NSString stringWithFormat:@"%@_simulator", simulatorName];
+  }
+
+  size_t size;
+  sysctlbyname("hw.model", NULL, &size, NULL, 0);
+  char* answer = malloc(size);
+  sysctlbyname("hw.model", answer, &size, NULL, 0);
+
+  NSString* results = [NSString stringWithCString:answer encoding:NSUTF8StringEncoding];
+  free(answer);
+  return results;
+}
+
+- (BOOL)compareImage:(UIImage*)a toOther:(UIImage*)b {
+  CGImageRef imageRefA = [a CGImage];
+  CGImageRef imageRefB = [b CGImage];
+
+  NSUInteger widthA = CGImageGetWidth(imageRefA);
+  NSUInteger heightA = CGImageGetHeight(imageRefA);
+  NSUInteger widthB = CGImageGetWidth(imageRefB);
+  NSUInteger heightB = CGImageGetHeight(imageRefB);
+
+  if (widthA != widthB || heightA != heightB) {
+    return NO;
+  }
+  NSUInteger bytesPerPixel = 4;
+  NSUInteger size = widthA * heightA * bytesPerPixel;
+  unsigned char* rawA = (unsigned char*)calloc(size, sizeof(unsigned char));
```

Comment:
There might be some better ways to compare images than comparing raw char* (some in here https://cate.blog/2015/01/21/creating-test-images-and-comparing-uiimages/ or maybe comparing UIImagePNGRepresentation?) but I'm guessing this is pretty performant as a tradeoff for being.... yeah.

Reply:
I we feel calloc-free is icky. `+[NSMutableData dataWithLength:]` should give you an autoreleased buffer that is zero initialized. Either way if fine but that might make it harder to forget to free these buffers on early returns.

Reply:
I don't want to compare the PNG bytes - I was getting false negatives that way in local tests, and I suspect that different versions of the OS/devices will end up decoding or encoding slightly differently.  If there's a better way to get the raw pixels I'm happy to do that.

Reply:
How would I pass the NSMutableData in?

Reply:
Oh I see..  I'll do that instead, definitely better.